scraper/main.go

```diff
@@ -142,6 +150,124 @@ type Logs struct {
 	Errors     []error
 }
 
+func test(clientConfig models.ClientConfig, objectType string, logsCh chan Logs, wg *sync.WaitGroup) {
+	defer wg.Done()
+	var initConf = `	
+terraform {
+	required_providers {
+		itsi = {
+			source  = "TiVo/splunk-itsi"
+			version = "~> 1.0"
+		}
+	}
+	required_version = "~> 1.7.0"
+}
+
+provider "itsi" { 
+{{if .BearerToken}}
+	smm_token = "{{.BearerToken}}"
+{{else}}
+	password = "{{.Password}}"
+	user     = "{{.User}}"
+{{end}}
+	host     = "{{.Host}}"
+	port     = {{.Port}}
+}`
+
+	var importSchema = `
+import {
+	id = "{{.Id}}"
+	to = {{.ObjectType}}.{{escape .TFID}}
+}
+`
+	base := models.NewBase(clientConfig, "", "", objectType)
+	errors := []error{}
+
+	importT := template.Must(template.New("import").Funcs(template.FuncMap{
+		"escape": provider.Escape,
+	}).Parse(importSchema))
+	initT := template.Must(template.New("init").Parse(initConf))
+
+	folder := fmt.Sprintf("%s/%s", outputBasePath, objectType)
+	err := os.MkdirAll(folder, os.ModePerm)
+	if err != nil {
+		panic(err)
+	}
+	import_fn := fmt.Sprintf("%s/import.tf", folder)
+	versions_fn := fmt.Sprintf("%s/versions.tf", folder)
+	versionsF, err := os.OpenFile(versions_fn, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer versionsF.Close()
+	initT.Execute(versionsF, clientConfig)
+
+	importF, err := os.OpenFile(import_fn, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer importF.Close()
+
+	cmd := exec.Command("terraform", "init")
+	cmd.Stdout = os.Stdout
+	cmd.Dir = folder
+	err = cmd.Run()
+	if err != nil {
+		panic(err)
+	}
+
+	for count, offset := base.GetPageSize(), 0; offset >= 0; offset += count {
+		ctx := context.Background()
+		items, err := base.Dump(ctx, &models.Parameters{Offset: offset, Count: count, Fields: []string{"_key", "title"}})
+		if err != nil {
+			errors = append(errors, err)
+			break
+		}
+		for _, item := range items {
+			err = importT.Execute(importF, map[string]string{
+				"Id":   item.RESTKey,
+				"TFID": item.TFID,
+				// TODO: this mapping is resource knowledge. implement mechanism to get via one call
+				"ObjectType": "itsi_" + objectType,
```

Comment:
I think `"itsi_" + objectType` allows matches the right resource name, so that's fine, otherwise we may have to add that info in the base metadata.